### DISCUSSION TITLE
High-level questions, to help user take decisions

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/bati#86] [Misc]
> Is your website internal?

Yes => auto-select `SPA`. (FYI :[`ssr` config](https://github.com/brillout/vike-react/issues/3).)

> Does the content of your website change only occasionally?

Yes => auto-select `Pre-rendering` and `GitHub Pages` / `Netlify`.

> Do you need a database?

Auto-select accordingly.

A neat thing is that it would enable us to show warnings if the user's selection doesn't make sense and/or isn't recommended.

**Comment by magne4000** [comment]
It could be a good alternative entry point for people which are not sure of what technology they want/need.
We probably should define a global flow to choose which Bati UI to show on which conditions.

**Comment by brillout** [comment]
I can't think of some kind of flow right now but maybe there will be some, e.g. if "No database" then some questions will become irrelevant. We'll see I guess 👀.

Maybe there are some questions we want to always show the user? For expert users it could be pesty, but I'm thinking a solution could be to make answering questions optional, so that expert users can go right ahead and select the tools they want.

As for the UI, maybe we can show the questions on a compact left sidebar? (If we don't need some kind of flow in the foreseeable future.)

> **Questions helping you choose:**
>  - <sub>Do you like JSX? [**[i]**]()</sub>
>  - <sub>Is your website internal? [**[i]**]()</sub>
>  - <sub>Does content change only occasionally? [**[i]**]()</sub>
>  - <sub>Do you need to persist data? [**[i]**]()</sub>

With compact yes/no toggles like on mobile UIs.

On a tangent, it could be neat to show a confirmation:

> ✅ **No issues detected.**

> ⚠️ **2 warnings:**
>   1. You selected `I like JSX` and `Vue`. While Vue does support JSX, it's a second class citizen and we recommend using Solid or React instead.
>   1. You selected `Content changes only occasionally` and `Vercel`. For websites with content that changes only occasionally (e.g. a few times a day), we recommend using a static host such as GitHub Pages or Netlify.

**Comment by magne4000** [comment]
> Maybe there are some questions we want to always show the user? For expert users it could be pesty, but I'm thinking a solution could be to make answering questions optional, so that expert users can go right ahead and select the tools they want.

A wizard UI would be for Junior devs and new Bati users, and the current UI for users that have already used it before I think. So I think that both UI will need to exist at any moment.

> As for the UI, maybe we can show the questions on a compact left sidebar? (If we don't need some kind of flow in the foreseeable future.)

That's something I would need to test

> On a tangent, it could be neat to show a confirmation

That's a good addition, could be added rather sooner than later if it's simple enough 👍

**Comment by redbar0n** [comment]
Not the be-all-end-all, but maybe some inspiration:


![image](https://github.com/batijs/bati/assets/459282/7c3fb787-47ea-4af3-8256-e6e7b6656a7a)
